arcgis-ios-sdk-samples/Scenes/Realistic lighting and shadows/RealisticLightingAndShadowsViewController.swift

```diff
@@ -0,0 +1,119 @@
+    let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.locale = .current
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .short
+        formatter.timeZone = .current
+        return formatter
+    }()
+    
+    // MARK: Instance methods
+    
+    /// Create a scene.
+    ///
+    /// - Returns: A new `AGSScene` object.
+    func makeScene() -> AGSScene {
+        // Create a scene layer from buildings REST service.
+        let buildingsURL = URL(string: "https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_BuildingShells/SceneServer")!
+        let buildingsLayer = AGSArcGISSceneLayer(url: buildingsURL)
+        // Create an elevation source from Terrain3D REST service.
+        let elevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        let elevationSource = AGSArcGISTiledElevationSource(url: elevationServiceURL)
+        let surface = AGSSurface()
+        surface.elevationSources = [elevationSource]
+        let scene = AGSScene(basemap: .topographic())
+        scene.baseSurface = surface
+        scene.operationalLayers.add(buildingsLayer)
+        return scene
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func sliderValueChanged(_ sender: UISlider) {
+        dateComponents.minute = Int(sender.value)
+        let date = Calendar.current.date(byAdding: dateComponents, to: today)!
+        dateTimeLabel.text = dateFormatter.string(from: date)
+        sceneView.sunTime = date
+    }
+    
+    @IBAction func modeButtonTapped(_ button: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose a lighting mode for the scene view.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        let modes: [(name: String, mode: AGSLightingMode)] = [
+            ("Light and shadows", .lightAndShadows),
+            ("Light only", .light),
+            ("No light", .noLight)
+        ]
+        modes.forEach { (name, mode) in
+            let action = UIAlertAction(title: name, style: .default) { _ in
+                self.sceneView.sunLighting = mode
+            }
+            alertController.addAction(action)
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = button
+        present(alertController, animated: true)
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["RealisticLightingAndShadowsViewController"]
+        sceneView.scene?.load { [weak self] (error) in
```

Comment:
I would suggest creating a separate `loadScene` function and call it from the `sceneView` observer if possible to simply `viewDidLoad`